### ISSUE TITLE
Space key twice `"  "` should become `". "` in non-buffer editor fields (like macOS)

### ISSUE THREAD

**Issue by fredrikekelund** [zed-industries/zed#30735]
### Summary

In text fields on macOS, tapping the space key twice in quick succession normally inserts a period ("."). This is not the case in the agent panel text area in Zed. The same goes for the git panel commit message text area. 

### Description

Steps to trigger the problem:
1. Open the Agent panel
2. Type `hello world` into the text area and then quickly hit the space key twice

Actual Behavior: Two spaces are inserted
Expected Behavior: One period and one space is inserted

### Zed Version and System Specs

Zed: v0.186.8 (Zed) 
OS: macOS 15.5.0
Memory: 16 GiB
Architecture: aarch64

**Comment by JosephTLyons** [comment]
Hey @fredrikekelund, I haven't investigated what other agent-powered editors do. Curious, are you coming from another one where they do this?

**Comment by fredrikekelund** [comment]
I am indeed coming from Cursor, where this behavior is implemented. 

However, I should clarify that this is the system standard for text fields on macOS. The comment text field that I'm typing in right now (here on GitHub) supports it. TextEdit, Messages, GitHub Desktop, and other apps also support it.

**Comment by JosephTLyons** [comment]
Gotcha, I know it's a standard on a lot of macOS apps and websites and such, but didn't realize any text editors implemented it for some fields. Thanks for letting me know!

**Comment by danilo-leal** [comment]
It actually seems like this is a general Zed missing feature, not specifically agent-related; it doesn't work generally in text areas that are not buffers. Part of me wonders if we really need to support this, but maybe we should because it seems like the baseline expectation of most text surfaces (that aren't a code editor). That said, I'll remove the `agent` label and edit the title and description.

**Comment by notpeter** [comment]
On macOS, this is optionally enabled and exposed as `NSAutomaticPeriodSubstitutionEnabled` in NSUserDefaults. Ideally Zed should respect the user's OS preference.  This is GUI accessible under System Preferences -> Keyboard -> "Edit" (next to Input Sources) -> "Add Period with Double Space"

I think this is ultimately part of a larger set of OS native control integrations which Zed does currently not support: Text Replacements: e.g. `omw` becomes `On my way!`, auto-capitalization, spelling correction, inline predictive text, etc.

Even if some of those are awful in a coding editor like "smart quotes and dashes" there are potentially accessibility reasons to consider supporting others.

<img width="436" height="446" alt="Image" src="https://github.com/user-attachments/assets/981397fc-c074-4231-9876-0b54c874fa8d" />

**Comment by fredrikekelund** [comment]
I can still reproduce it in the latest version of Zed
